contrib/database/sql/conn.go

```diff
@@ -253,7 +253,16 @@ func (tc *TracedConn) injectComments(ctx context.Context, query string, mode tra
 	if span, ok := tracer.SpanFromContext(ctx); ok {
 		spanCtx = span.Context()
 	}
-	carrier := tracer.SQLCommentCarrier{Query: query, Mode: mode, DBServiceName: tc.cfg.serviceName, PeerDBHostname: tc.meta[ext.TargetHost], PeerDBName: tc.meta[ext.DBName]}
+
+	// This occurs if the user sets peer.service explicitly while creating the connection
+	var peerServiceName string
+	if meta, ok := ctx.Value(spanTagsKey).(map[string]string); ok {
+		if peerServiceTag, ok := meta[ext.PeerService]; ok {
+			peerServiceName = peerServiceTag
+		}
+	}
```

Comment:
I think you should also check for tags coming from `tc.cfg.tags`, which could be set using the `WithCustomTag` option, which can be set either in the driver `sqltraceRegister` and/or  `sqltrace.Open` (this would be less specific - so less priority than the ones you added above, which come from the context in specific db operations)